Dart/src/com/jetbrains/lang/dart/ide/toolingDaemon/DartToolingDaemonService.kt

```diff
@@ -256,6 +306,16 @@ class DartToolingDaemonService private constructor(private val project: Project)
         val params = json["params"].asJsonObject
         val streamId = params["streamId"].asString
         eventDispatcher.multicaster.received(streamId, json)
+      } else if (servicesMap.contains(method)) {
+        val params = json["params"].asJsonObject
+        val id = json["id"].asString
+        val consumer = servicesMap.get(method)
+        val result = consumer?.received(params)
+        if (result == null) {
+          logger.warn("No response from request consumer for message: $text")
+        } else {
+          sendResponse(id, result)
+        }
```

Comment:
I rewrote it in a way that neither `consumer`, nor `result` are nullable inside the `else if` branch:
```
      val method = json["method"]?.asString
      val serviceConsumer = servicesMap[method]
      if (method == "streamNotify") {...}
      else if (serviceConsumer != null) {
        val params = json["params"].asJsonObject
        val id = json["id"].asString
        val result = serviceConsumer.received(params)
        sendResponse(id, result)
      }
      else {...}
```
Is it ok?

Reply:
Looks great! Thanks